perfkitbenchmarker/benchmarks/speccpu2006_benchmark.py

```diff
@@ -186,16 +193,26 @@ def ExtractScore(stdout, vm):
       result_section.pop()
 
   metadata = {'machine_type': vm.machine_type, 'num_cpus': vm.num_cpus}
-  results.append(sample.Sample(spec_name, spec_score, '', metadata))
 
+  missing_results = False
   for benchmark in result_section:
-    # ignore failed runs
-    if re.search('NR', benchmark):
+    # Skip over failed runs, but count them since they make the overall
+    # result invalid.
+    if 'NR' in benchmark:
+      missing_results = True
       continue
     # name, ref_time, time, score, misc
     name, _, _, score, _ = benchmark.split()
     results.append(sample.Sample(str(name), float(score), '', metadata))
 
+  if missing_results:
+    if FLAGS.runspec_keep_partial_results:
+      spec_name += '_partial'
```

Comment:
After further discussion, consensus was that dashboards that use 'contains' should be fixed instead of doing more complex name mangling here.